DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Fields/ContentPickerFieldGraphSyncer.cs

```diff
@@ -24,17 +27,67 @@ public class ContentPickerFieldGraphSyncer : IContentFieldGraphSyncer
         private const string ContentItemIdsKey = "ContentItemIds";
         //todo: move into hidden ## section?
         private static readonly Regex _relationshipTypeRegex = new Regex("\\[:(.*?)\\]", RegexOptions.Compiled);
-        private readonly ILogger<ContentPickerFieldGraphSyncer> _logger;
-
+        private readonly IPreExistingContentItemVersion _preExistingContentItemVersion;
+        private readonly IContentDefinitionManager _contentDefinitionManager;
         public const string ContentPickerRelationshipPropertyName = "contentPicker";
 
         public static IEnumerable<KeyValuePair<string, object>> ContentPickerRelationshipProperties { get; } =
             new Dictionary<string, object> { { ContentPickerRelationshipPropertyName, true } };
 
         public ContentPickerFieldGraphSyncer(
-            ILogger<ContentPickerFieldGraphSyncer> logger)
+            IPreExistingContentItemVersion preExistingContentItemVersion,
+            IContentDefinitionManager contentDefinitionManager)
         {
-            _logger = logger;
+            _preExistingContentItemVersion = preExistingContentItemVersion;
+            _contentDefinitionManager = contentDefinitionManager;
+        }
+
+        public async Task AddRelationship(IDescribeRelationshipsContext parentContext)
+        {
+            ContentPickerFieldSettings contentPickerFieldSettings =
+               parentContext.ContentPartFieldDefinition!.GetSettings<ContentPickerFieldSettings>();
+
+            JArray? contentItemIdsJArray = (JArray?)parentContext.ContentField![ContentItemIdsKey];
+
+            if (contentItemIdsJArray != null && contentItemIdsJArray.Count > 0)
+            {
+                string relationshipType = await RelationshipTypeContentPicker(contentPickerFieldSettings, parentContext.SyncNameProvider);
+                var sourceNodeLabels = await parentContext.SyncNameProvider.NodeLabels(parentContext.ContentItem.ContentType);
+                string pickedContentType = contentPickerFieldSettings.DisplayedContentTypes[0];
+                IEnumerable<string> destNodeLabels = await parentContext.SyncNameProvider.NodeLabels(pickedContentType);
+                parentContext.AvailableRelationships.Add(new ContentItemRelationship(sourceNodeLabels, relationshipType, destNodeLabels));
+
+                await GetRelationshipsForNestedContentPickers(parentContext, contentItemIdsJArray);
```

Comment:
This needs to call back to get the relationships at the item level.
For an example, see `EmbeddedContentItemsGraphSyncer` calling back using `SyncEmbedded` or `DeleteEmbedded`.
The item level code in this syncer is only looking for content pickers within the embedded content items, but it needs to handle all parts that describe relationships, such as containing/embedding parts.

Reply:
I think I can just use the BuildRelationships command and pass in the retrieved content item here. BuildCommands returns the actual Neo commands. I have moved it into it's own class anyway for SRP